### DISCUSSION TITLE
400 - 'Summary or Text is required.' Using Jenkins webhook

### DISCUSSION THREAD

**Discussion by Ravikanthreddy** [test-results-reporter/reporter#112] [Q&A] [answered]
Hi,
First of all thanks for this plugin. Super useful if the below issue gets resolved. For some reason i cannot use incoming webhook. I have to use Jenkins webhook instead, think disabled by my Org the incoming one.

When i use the default config with Junit xml i get the below error.
StatusCodeError: 400 - Summary or Text is required. Tried adding the summary parameter in the json but without success.
```json
{
    "reports": [
      {
        "targets": [
          {
            "name": "teams",
            "inputs": {
              "url": "Webhook url", 
              "publish": "test-summary-slim"
            }
          }
        ],
        "results": [
          {
            "type": "junit",
            "files": ["resources/newman/report.xml"]
          }
        ]
      }
    ]
}
```

**Comment by ASaiAnudeep** [comment]
can you share the contents of report.xml

**Reply by Ravikanthreddy** [reply]
```xml 
 <?xml version="1.0" encoding="UTF-8"?>
  <testsuites name="test-run" tests="1" time="1.471">
    <testsuite name="Market Cases / Process Payment / sample test"  timestamp="2023-01-12T12:15:37.522Z" tests="2" failures="1" errors="0" time="1.471">
      <testcase name="Status code is 200" time="1.471" classname="testrun"/>
      <testcase name="Verify user is logged in" time="1.471" classname="testrun">
        <failure type="AssertionFailure" message="Cannot read property 'isLoggedIn' of undefined">
          <![CDATA[Failed 1 times.]]>
          <![CDATA[Collection JSON ID: 38edd670-8f3e-4924-b8a4-6dfe2f1f7a92.]]>
          <![CDATA[Collection name: test-run.]]>
          <![CDATA[Request name: Market Cases /  Process Payment /  sammple test]]>
          <![CDATA[Test description: Verify user is logged in.]]>
          <![CDATA[Error message: Cannot read property 'isLoggedIn' of undefined.]]>
          <![CDATA[Stacktrace: TypeError: Cannot read property 'isLoggedIn' of undefined
    at Object.eval sandbox-script.js:2:1).]]>
        </failure>
      </testcase>
    </testsuite>
  </testsuites>
  ```

**Reply by Ravikanthreddy** [reply]
@ASaiAnudeep Managed to get somehow the incoming webhook connector and it works fine. So issue was only with Jenkins connector as i was mentioning already. Thank you so much for the plugin. :)

**Comment by yasafooo** [comment]
I'm also getting the same error. Please help asap

`{
    "reports": [
      {
        "targets": [
          {
            "name": "teams",
            "inputs": {
              "url": "url"
            }
          }
        ],
        "results": [
          {
            "type": "junit",
            "files": ["test-results/testresult.xml"]
          }
        ]
      }
    ]
  }`

```
<testsuites id="" name="" tests="2" failures="0" skipped="0" errors="0" time="5.750959999978543">
<testsuite name="example.spec.js" timestamp="1674187440763" hostname="" tests="2" failures="0" skipped="0" time="0.086" errors="0">
<testcase name="Administration - Verify Company Profile Test  Home Page" classname="[chromium] › example.spec.js › Administration - Verify Company Profile › Test  Home Page" time="0.02">
<system-out>
<![CDATA[Finished Test  Home Page with status passed
]]>
</system-out>
</testcase>
<testcase name="Administration - Verify Company Profile Test  Get Started Page" classname="[chromium] › example.spec.js › Administration - Verify Company Profile › Test  Get Started Page" time="0.066">
<system-out>
<![CDATA[Finished Test  Get Started Page with status passed
After tests
]]>
</system-out>
</testcase>
</testsuite>
</testsuites>
```

**Reply by ASaiAnudeep** [reply]
hello @Ravikanthreddy, can you help @yasafooo to resolve the issue and also mention the steps here. Thanks in Advance.

**Comment by Ravikanthreddy** [comment, marked as answer]
There is no issue with plugin to be precise. There are two connectors with teams 1. Jenkins webhook connector 2. Incoming webhook.
Plugin works perfectly fine with incoming webhook. Jenkins connector expects a different payload or bit more specific like adaptive cards etc and plugin gives error with that.
In my case, luckily my org enabled incoming webhook and that solved my issue.
hope this clarifies @yasafooo
